doxia-core/src/main/java/org/apache/maven/doxia/index/IndexingSink.java

```diff
@@ -69,6 +72,10 @@
     /** The stack. */
     private final Stack<IndexEntry> stack;
 
+    /** A map containing all used ids of index entries as key and how often they are used as value
+     * (0-based, i.e. 0 means used 1 time). */
+    private final Map<String, AtomicInteger> usedIds;
```

Comment:
Why atomic? All you need is a mutuable integer, no? This component is neither singleton not in MT use.

Reply:
Which MutableInteger implementation should I use? I don't want to implement it myself and the performance drawback is IMHO neglectable here.

Reply:
Ideally from Commons Lang, but we don't have it in the class path. Since for a single class it doesn't really make sense, let's stick to it and please add a comment which depicts that we explicitly don't require the atomic semantics, but the mutuability only.

Reply:
Done in https://github.com/apache/maven-doxia/pull/106/commits/41ad9261b547cfeda7f8cb7ac5827f18fcf0f598.